common/p_enemy.cpp

```diff
@@ -2787,12 +2787,14 @@ void A_BossDeath(AActor *actor)
 			return;
 		}
 	}
-	else if (level.flags & LEVEL_CYBORGSPECIAL && actor->flags3 & MF3_E4M6BOSS)
+	else if (level.flags & LEVEL_CYBORGSPECIAL && actor->flags3 & MF3_E4M6BOSS &&
+	         level.mapname == "E4M6")
 	{
 		EV_DoDoor(DDoor::doorOpen, NULL, NULL, 666, SPEED(64), 0, NoKey);
 		return;
 	}
-	else if (level.flags & LEVEL_SPIDERSPECIAL && actor->flags3 & MF3_E4M8BOSS)
+	else if (level.flags & LEVEL_SPIDERSPECIAL && actor->flags3 & MF3_E4M8BOSS &&
+	         level.mapname == "E4M8")
```

Comment:
I'm not sure the right play is hardcoding the map name.  Is there no way to do this otherwise?

Reply:
Nevermind, I see what's going on.  How big of a pain in the neck would it be to add support for the following MAPINFO tokens?

- SpecialAction_ExitLevel
- SpecialAction_OpenDoor
- SpecialAction_LowerFloor
- SpecialAction_KillMonsters 

Seems like it just be a couple of one-note tokens that set a flag.